### DISCUSSION TITLE
User registration

### DISCUSSION THREAD

**Discussion by martinbndr** [pelican-dev/panel#17] [Ideas]
Users should be able to create panel accounts themself, but in admin panel it should be toggled on/off.

**Comment by lancepioch** [comment]
It doesn't make sense to me that any person should be able to create an account on your panel without you specifically already inviting them. Could you give me a use case?

**Reply by wdhdev** [reply]
Some hosts have plugins and such so users can create their own servers on the client UI, so allowing users to just register on the panel and not requiring an admin or a bot to do it would be quite helpful.

**Reply by martinbndr** [reply]
Would be cool to have it to allow users self register as some use cases like on personal use, friends can just create an account without having to wait for an admin.
Also at some hosting providers it's just useful not having to build a own frontent or application allowing them to self register via the application API.

Basic login stuff including registration should be added to the core code maintained by its core developers.
As I said it should be configurable through the admin dashboard to respect all users needs to deactivate that feature.

**Reply by lancepioch** [reply]
I think it'd be a huge security issue if it was possible for a user to register themselves and already become a full/root administrator. Bots and scripts are constantly scanning every single public ip address. Once this becomes possible for the panel, all it will take is less than a minute of your panel being open for registration, and a bot could create itself as a root admin and immediately compromise your system.

I am not convinced that it's a barrier if you just need to invite somebody by their email (where they get a random pw). Alternatively you can create an account and set up their password beforehand and message it to them. This takes no time at all.

**Reply by BroOtti** [reply]
Why should a fresh user get full access? You will still need to configure permission for a new user. But thinking of a discord integration, certain permission could be linked to a discord role. So, If the user has a configured discord role they will be able to manage a Server or something Like that. That way you don't have to create an user for them and they can just login using Discord.

**Comment by Bocon778** [comment]
I feel like it would be best for this to be kept as a plugin option, because not everyone wants users to register themselves.

**Reply by martinbndr** [reply]
It's such a basic thing in a web application like this to login /register for an account.
It's just the bad practise to have this basic feature in a plugin, it could just be added to the core code and maintained by the core development team.
Does not mean users always able to register, that's why I also suggest, making a configuration to enable/disable that feature.

**Comment by lancepioch** [comment]
This can be done as a plugin, sorry it doesn't fit in with the majority use case. The majority of people are going to set their panel up, then invite a friend or family member or a handful of buddies.
